File: tests/bun/tests/simple.ts
export function makeTetradicNumber(index: number): number {
  return (index * (index + 1) * (index + 2)) / 6;
}
const makeTetradicNumbers = (num: number) => {
  const numbers = []
  let index = 0

  while (true) {
    const tetradicNumber = makeTetradicNumber(index);
    if (tetradicNumber > num) {
      break;
    }

    numbers.push(tetradicNumber);
    index++
  }

  return numbers;
}

function findSums(inputNumber: number): number[] | null {
  const tetradicNumbers = makeTetradicNumbers(inputNumber);
  const length = tetradicNumbers.length;

  for (let i1 = 0; i1 < length; i1++) {
    const n1 = makeTetradicNumber(i1);
    const required1 = inputNumber - n1;
    for (let i2 = 0; i2 < length; i2++) {
      const n2 = makeTetradicNumber(i2);
      const required2 = required1 - n2;
      for (let i3 = 0; i3 < length; i3++) {
        const n3 = makeTetradicNumber(i3);
        const required3 = required2 - n3;
        for (let i4 = 0; i4 < length; i4++) {
          const n4 = makeTetradicNumber(i4);
          const required4 = required3 - n4;
          for (let i5 = 0; i5 < length; i5++) {
            const n5 = makeTetradicNumber(i5);
            const required5 = required4 - n5;
            if (required5 === 0) {
              return [n1, n2, n3, n4, n5];
            }
          }
        }
      }
    }
  }

  return null;
}

function test(start: number, end: number, step: number) {
  for (let i = start; i < end; i++) {
    const result = findSums(i);

    if (result === null || result.reduce((acc, curr) => acc + curr, 0) !== i) {
      throw new Error(`No result for ${i}`);
    }
  }
}

const args = Bun.argv.slice(2).map(Number) as [number, number, number];

test(...args);

File: tests/python/tests/simple.py
import sys

def make_tetradic_number(index: int) -> int:
  return (index * (index + 1) * (index + 2)) // 6

def make_tetradic_numbers(num: int) -> list:
  numbers = []
  index = 0

  while True:
    tetradic_number = make_tetradic_number(index)
    if tetradic_number > num:
      break

    numbers.append(tetradic_number)
    index += 1

  return numbers

def find_sums(input_number: int) -> list:
  tetradic_numbers = make_tetradic_numbers(input_number)
  length = len(tetradic_numbers)

  for i1 in range(length):
    n1 = make_tetradic_number(i1)
    required1 = input_number - n1
    for i2 in range(length):
      n2 = make_tetradic_number(i2)
      required2 = required1 - n2
      for i3 in range(length):
        n3 = make_tetradic_number(i3)
        required3 = required2 - n3
        for i4 in range(length):
          n4 = make_tetradic_number(i4)
          required4 = required3 - n4
          for i5 in range(length):
            n5 = make_tetradic_number(i5)
            required5 = required4 - n5
            if required5 == 0:
              return [n1, n2, n3, n4, n5]

  return None

def test(start: int, end: int, step: int):
  for i in range(start, end):
    result = find_sums(i)

    if result is None or sum(result) != i:
      raise ValueError(f"No result for {i}")

if __name__ == "__main__":
  args = list(map(int, sys.argv[1:4]))
  test(*args)